client/src/components/Map/Map.tsx

```diff
@@ -0,0 +1,174 @@
+  }
+
+  const onAutosuggestClick = async (e: any, location: any) => {
+    e.preventDefault();
+    if (location.weergavenaam) {
+      locationRef.current.value = location.weergavenaam;
+      dispatch({
+        type: 'setShowAutosuggest',
+        payload: {
+          showAutosuggest: false
+        }
+      });
+    }
+
+    const response = await axios.get(`https://geodata.nationaalgeoregister.nl/locatieserver/v3/lookup?id=${location.id}`)
+    if (mapInstance && response.data.response.docs[0]) {
+      const loc = response.data.response.docs[0].centroide_ll.replace(/POINT\(|\)/, '').split(' ');
+      const flyTo = { lat: parseFloat(loc[1]), lng: parseFloat(loc[0]) };
+      mapInstance.flyTo(flyTo, 11);
+      dispatch({
+        type: 'setLatLng',
+        payload: {
+          latLng: flyTo
+        }
+      });
+    }
+  }
+
+  useEffect(() => {
+    fetchAutosuggest(url);
+  }, [url]);
+
+  useEffect(() => {
+    if (location && location.length) {
+      locationRef.current.value = location[0].weergavenaam;
+    } else if (location && location.length === 0) {
+      locationRef.current.value = '';
+    }
+  }, [location, locationRef]);
+
+  const handleClickOutside = event => {
+    if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
+      dispatch({
+        type: 'setShowAutosuggest',
+        payload: {
+          showAutosuggest: false
+        }
+      });
+
+      locationRef.current.value = '';
+    }
+  };
+  useEffect(() => {
+    document.body.addEventListener('click', handleClickOutside, false)
+  }, []);
+
+  return (
+    <div style={{ position: 'relative', width: '100%', height: '500px' }}>
+      <StyledMap setInstance={(instance) => {
+          dispatch({
+            type: 'setMapInstance',
+            payload: {
+              mapInstance: instance
+            }
+          });
+      }}
+        events={{
+          click: (e) => {
+            onMapclick(e);
+          },
+        }}
+      >
+        {latLng && <Marker latLng={latLng} />}
+        <ViewerContainer
+          bottomRight={<Zoom />}
+          topLeft={
+            <>
+              <Input
+                style={{ width: '500px' }}
+                id="location"
+                ref={locationRef}
+                onChange={(e) => {
+                  dispatch({
+                    type: 'onChangeLocation',
+                    payload: {
+                      query: e.target.value,
+                      url: `${autosuggestUrl}${e.target.value}`
+                    }
+                  });
+
+                  // setQuery(e.target.value);
+                  // setUrl(`${autosuggestUrl}${e.target.value}`);
+                }}
+              />
+              {showAutosuggest &&
+                <ul ref={wrapperRef} style={{ width: '500px', backgroundColor: 'white', listStyleType: 'none', padding: '0 0 0 12px' }}>
+                  {query.length && autosuggest && autosuggest.length ? autosuggest.map((location) =>
```

Comment:
In plaats van een ternary (? <li...> : null) zou je ook && kunnen gebruiken. 
Dus `query.length && autosuggest && autosuggest.length && <li>`